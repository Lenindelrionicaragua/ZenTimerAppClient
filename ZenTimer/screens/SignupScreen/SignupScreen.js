import React, { useState } from "react";
import { StatusBar } from "react-native";
import KeyboardAvoider from "../../component/KeyboardAvoider/KeyboardAvoider";
import DateTimePicker from "@react-native-community/datetimepicker";
import { Formik } from "formik";
import {
  StyledContainer,
  InnerContainer,
  PageTitle,
  SubTitle,
  StyledFormArea,
  StyledButton,
  ButtonText,
  MsgBox,
  Line,
  FooterView,
  FooterText,
  FooterLink,
  FooterLinkContent,
} from "./SignupScreenStyles";
import { Colors } from "../../styles/AppStyles";
import TextInputSignupScreen from "../../component/TextInputSignupScreen/TextInputSignupScreen";

// Colors
const { lightGrey } = Colors;

const SignupScreen = ({ navigation }) => {
  const [hidePassword, setHidePassword] = useState(true);
  const [show, setShow] = useState(false);
  const [date, setDate] = useState(new Date(2000, 0, 1));
  const [userBirthDay, setUserBirthDay] = useState();

  const onChange = (event, selectedDate) => {
    const currentDate = selectedDate || date;
    setShow(false);
    setDate(currentDate);
    setUserBirthDay(currentDate);
  };

  const showDatePicker = () => {
    setShow(true);
  };

  return (
    <KeyboardAvoider>
      <StyledContainer testID="styled-container">
        <StatusBar style="dark" />
        <InnerContainer testID="inner-container">
          <PageTitle testID="page-title">ZenTimer</PageTitle>
          <SubTitle testID="sub-title">Account Sign Up</SubTitle>

          {show && (
            <DateTimePicker
              testID="date-time-picker"
              value={date}
              mode="date"
              is24Hour={true}
              display="default"
              onChange={onChange}
            />
          )}

          <Formik
            initialValues={{
              fullName: "",
              email: "",
              dateOfBirth: "",
              password: "",
              confirmPassword: "",
            }}
            onSubmit={(values) => {
              console.log(values);
            }}
          >
            {({ handleChange, handleBlur, handleSubmit, values }) => (
              <StyledFormArea>
                <TextInputSignupScreen
                  label="Full Name"
                  icon="person"
                  placeholder="Zen User"
                  placeholderTextColor={lightGrey}
                  onChangeText={handleChange("fullName")}
                  onblur={handleBlur("fullName")}
                  value={values.fullName}
                  testID="full-name"
                />

                <TextInputSignupScreen
                  label="Email Address"
                  icon="mail"
                  placeholder="serenity@gmail.com"
                  placeholderTextColor={lightGrey}
                  onChangeText={handleChange("email")}
                  onblur={handleBlur("email")}
                  value={values.email}
                  keyboardType="email-address"
                  testID="email-input"
                />

                <TextInputSignupScreen
                  label="Date of Birth"
                  icon="calendar"
                  placeholder="YYYY - MM - DD"
                  placeholderTextColor={lightGrey}
                  onChangeText={handleChange("dateOfBirth")}
                  onblur={handleBlur("dateOfBirth")}
                  value={userBirthDay ? userBirthDay.toDateString() : ""}
                  testID="date-of-birth"
                  isDate={true}
                  editable={false}
                  showDatePicker={showDatePicker}
                />

                <TextInputSignupScreen
                  label="Password"
                  icon="lock"
                  placeholder="* * * * * * *"
                  placeholderTextColor={lightGrey}
                  onChangeText={handleChange("password")}
                  onblur={handleBlur("password")}
                  value={values.password}
                  secureTextEntry={hidePassword}
                  testID="password-input"
                  isPassword={true}
                  hidePassword={hidePassword}
                  setHidePassword={setHidePassword}
                />

                <TextInputSignupScreen
                  label="Confirm Password"
                  icon="lock"
                  placeholder="* * * * * * *"
                  placeholderTextColor={lightGrey}
                  onChangeText={handleChange("confirmPassword")}
                  onblur={handleBlur("confirmPassword")}
                  value={values.confirmPassword}
                  secureTextEntry={hidePassword}
                  testID="confirm-password-input"
                  isPassword={true}
                  hidePassword={hidePassword}
                  setHidePassword={setHidePassword}
                />
                <MsgBox testID="msg-box">...</MsgBox>
                <StyledButton
                  testID="login-styled-button"
                  onPress={handleSubmit}
                >
                  <ButtonText testID="login-button-text">Sign Up</ButtonText>
                </StyledButton>
                <Line testID="line" />
                <FooterView testID="footer-view">
                  <FooterText testID="footer-text">
                    Already have an account?
                  </FooterText>
                  <FooterLink testID="footer-link">
                    <FooterLinkContent testID="footer-link-content">
                      Login
                    </FooterLinkContent>
                  </FooterLink>
                </FooterView>
              </StyledFormArea>
            )}
          </Formik>
        </InnerContainer>
      </StyledContainer>
    </KeyboardAvoider>
  );
};

export default SignupScreen;
