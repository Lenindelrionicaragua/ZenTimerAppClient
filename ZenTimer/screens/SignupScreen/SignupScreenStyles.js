import styled from "styled-components/native";
import { View, Image, Text, TextInput, TouchableOpacity } from "react-native";
import Constants from "expo-constants";
import { Colors } from "../../styles/AppStyles";

const StatusBarHeight = Constants.statusBarHeight;

const { white, orange, grey, yellow, lightGrey, black } = Colors;

export const StyledContainer = styled(View)`
  flex: 1;
  padding: 25px;
  padding-top: ${StatusBarHeight + 10}px;
  background-color: ${lightGrey};
`;

export const InnerContainer = styled(View)`
  flex: 1;
  width: 100%;
  align-items: center;
`;

export const PageLogo = styled(Image)`
  width: 200px;
  height: 200px;
  border-radius: 100px;
  overflow: hidden;
`;

export const PageTitle = styled(Text)`
  font-size: 30px;
  text-align: center;
  font-weight: bold;
  color: ${black};
  padding: 10px;
`;

export const SubTitle = styled(Text)`
  font-size: 10px;
  margin-bottom: 20px;
  letter-spacing: 1px;
  font-weight: bold;
  color: ${grey};
`;

export const StyledFormArea = styled(View)`
  width: 90%;
`;

export const StyledTextInput = styled(TextInput)`
  background-color: ${orange};
  padding: 15px;
  padding-left: 55px;
  padding-right: 55px;
  border-radius: 5px;
  font-size: 16px;
  height: 60px;
  margin-vertical: 3px;
  margin-bottom: 10px;
  color: ${grey};
`;

export const StyledInputLabel = styled(Text)`
  color: ${grey};
  font-size: 13px;
  text-align: left;
`;

export const LeftIcon = styled(View)`
  left: 15px;
  top: 38px;
  position: absolute;
  z-index: 1;
`;

export const RightIcon = styled(TouchableOpacity)`
  right: 15px;
  top: 38px;
  position: absolute;
  z-index: 1;
`;

export const StyledButton = styled(TouchableOpacity)`
  padding: 15px;
  background-color: ${black};
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  margin-vertical: 5px;
  height: 60px;

  ${(props) =>
    props.google == true &&
    `
    background-color: ${orange};
    flex-direction: row;
    justify-content: center;
  `}
`;

export const ButtonText = styled(Text)`
  color: ${white};
  font-size: 16px;

  ${(props) =>
    props.google == true &&
    `
    padding: 25px;
    color: ${black}
  `}
`;

export const MsgBox = styled(Text)`
  text-align: center;
  font-size: 13px;
`;

export const Line = styled(View)`
  height: 1px;
  width: 100%;
  background-color: ${orange};
  margin-vertical: 10px;
`;

export const ExtraView = styled(View)`
  justify-content: center;
  flex-direction: row;
  align-items: center;
  padding: 10px;
`;

export const ExtraText = styled(Text)`
  justify-content: center;
  align-content: center;
  color: ${grey};
  font-size: 15px;
`;

export const TextLink = styled(TouchableOpacity)`
  justify-content: center;
  align-items: center;
`;

export const TextLinkContent = styled(Text)`
  color: ${yellow};
  font-size: 15px;
`;
