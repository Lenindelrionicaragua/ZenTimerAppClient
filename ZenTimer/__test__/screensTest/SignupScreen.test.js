import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, act, waitFor } from "@testing-library/react-native";

import DateTimePicker from "@react-native-community/datetimepicker";

import { Fontisto } from "@expo/vector-icons";

import SignupScreen, {
  MyTextInput,
} from "../../screens/SignupScreen/SignupScreen";

import { Formik } from "formik";
import { StatusBar } from "react-native";
import {
  StyledContainer,
  InnerContainer,
  PageLogo,
  PageTitle,
  SubTitle,
} from "../../screens/SignupScreen/SignupScreenStyles";

// Rendering Functions
const renderSignupScreen = () => render(<SignupScreen />);
const renderSignupScreenWithRenderer = () => renderer.create(<SignupScreen />);

let signupScreenRender;
let signupScreenRenderWithRenderer;

beforeEach(() => {
  signupScreenRender = renderSignupScreen();
  signupScreenRenderWithRenderer = renderSignupScreenWithRenderer();
});

//SignupScreen
describe("SignupScreen", () => {
  test("Renders correctly the SignupScreen Component", () => {
    const signupScreenSnapshot = signupScreenRenderWithRenderer.toJSON();
    expect(signupScreenSnapshot).toMatchSnapshot();
  });

  test("Render StatusBar correctly", () => {
    const signupScreenInstance = signupScreenRenderWithRenderer.root;
    const statusBar = signupScreenInstance.findByType(StatusBar);
    expect(statusBar).toBeTruthy();
  });

  test("Render UI components correctly", () => {
    const { getByTestId } = signupScreenRender;

    expect(getByTestId("styled-container")).toBeTruthy();
    expect(getByTestId("inner-container")).toBeTruthy();
    expect(getByTestId("page-title")).toBeTruthy();
    expect(getByTestId("sub-title")).toBeTruthy();
  });

  test("PageTitle should render a string of letters, numbers, or spaces", () => {
    const { getByTestId } = signupScreenRender;
    const pageTitleComponent = getByTestId("page-title");
    const textContent = pageTitleComponent.props.children.toString();
    expect(textContent).toMatch("ZenTimer");
  });

  test("SubTitle should render a string of letters, numbers or spaces", () => {
    const { getByTestId } = signupScreenRender;
    const subTitleComponent = getByTestId("sub-title");
    const textContent = subTitleComponent.props.children.toString();
    expect(textContent).toMatch("Account Signup");
  });
});

// Formik Integration Tests
describe("Formik Integration Tests", () => {
  let formikComponent;

  beforeEach(() => {
    const signupScreenInstance = signupScreenRenderWithRenderer.root;
    formikComponent = signupScreenInstance.findByType(Formik);
  });

  test("Render a Formik component", () => {
    expect(formikComponent).toBeTruthy();
  });

  test("Should have initialValues", () => {
    expect(formikComponent.props.initialValues).toEqual({
      fullName: "",
      email: "",
      dateOfBirth: "",
      password: "",
      confirmPassword: "",
    });
  });

  test("Should have a function as a child", () => {
    expect(typeof formikComponent.props.children).toBe("function");
  });

  //MyTextInput
  describe("MyTextInput", () => {
    let fullName;
    let emailInput;
    let dateOfBirth;
    let passwordInput;
    let confirmPasswordInput;

    const renderForm = () => {
      const { getByTestId } = signupScreenRender;
      fullName = getByTestId("full-name");
      emailInput = getByTestId("email-input");
      dateOfBirth = getByTestId("date-of-birth");
      passwordInput = getByTestId("password-input");
      confirmPasswordInput = getByTestId("confirm-password-input");
    };

    beforeEach(() => {
      renderForm();
    });

    describe("Rendering", () => {
      test("Renders correctly the full-name", () => {
        expect(fullName).toBeTruthy();
      });

      test("Renders correctly the email-input", () => {
        expect(emailInput).toBeTruthy();
      });

      test("Renders correctly the date-of-birth", () => {
        expect(dateOfBirth).toBeTruthy();
      });

      test("Renders correctly the password-input", () => {
        expect(passwordInput).toBeTruthy();
      });

      test("Renders correctly the confirm-password-input", () => {
        expect(confirmPasswordInput).toBeTruthy();
      });
    });

    describe("Form State Update", () => {
      let dateOfBirth;

      const renderForm = () => {
        const { getByTestId } = signupScreenRender;
        dateOfBirth = getByTestId("date-of-birth");
      };

      beforeEach(() => {
        renderForm();
      });

      beforeEach(() => {
        renderForm();
      });

      test("Correctly updates form state on onChangeText and onBlur", () => {
        act(() => {
          fireEvent.changeText(fullName, "Zen User");
          fireEvent(fullName, "blur");

          fireEvent.changeText(emailInput, "serenity@gmail.com");
          fireEvent(emailInput, "blur");

          fireEvent.changeText(dateOfBirth, "");
          fireEvent(dateOfBirth, "blur");

          fireEvent.changeText(passwordInput, "password123");
          fireEvent(passwordInput, "blur");

          fireEvent.changeText(confirmPasswordInput, "password123");
          fireEvent(confirmPasswordInput, "blur");
        });

        expect(fullName.props.value).toBe("Zen User");
        expect(emailInput.props.value).toBe("serenity@gmail.com");
        expect(dateOfBirth.props.value).toBe("");
        expect(passwordInput.props.value).toBe("password123");
        expect(confirmPasswordInput.props.value).toBe("password123");
      });
    });
  });

  describe("DateTimePicker", () => {
    let dateOfBirth;

    const renderForm = () => {
      const { getByTestId } = signupScreenRender;
      dateOfBirth = getByTestId("date-of-birth");
    };

    beforeEach(() => {
      renderForm();
    });

    test("Renders correctly the date-of-birth input field", () => {
      expect(dateOfBirth).toBeTruthy();
    });

    test("Show DateTimePicker after clicking on the date-of-birth input field", async () => {
      fireEvent.press(dateOfBirth);

      await waitFor(() => {
        const dateTimePicker =
          signupScreenRender.queryByTestId("date-time-picker");
        expect(dateTimePicker).toBeTruthy();
      });
    });

    test("Change date in DateTimePicker", async () => {
      fireEvent.press(dateOfBirth);

      await waitFor(() => {
        const dateTimePicker =
          signupScreenRender.queryByTestId("date-time-picker");
        expect(dateTimePicker).toBeTruthy();

        fireEvent(dateTimePicker, "onChange", {
          nativeEvent: { timestamp: "Tue Feb 01 2022" },
        });
      });

      await waitFor(() => {
        expect(dateOfBirth.props.value).toBe("Tue Feb 01 2022");
      });
    });
  });

  // Login StyledButton
  describe("StyledButton", () => {
    test("Render StyledButton", () => {
      const { getByTestId } = signupScreenRender;
      const styledButtonElement = getByTestId("login-styled-button");
      expect(styledButtonElement).toBeTruthy();
    });
  });

  // MsgBox
  describe("MsgBox", () => {
    test("Render a MsgBox", () => {
      const { getByTestId } = signupScreenRender;
      const msgBoxElement = getByTestId("msg-box");
      expect(msgBoxElement).toBeTruthy();
    });

    test("MsgBox should render a string of letters, numbers or spaces", async () => {
      const { getByTestId } = signupScreenRender;
      const msgBoxElement = getByTestId("msg-box");
      const textContent = msgBoxElement.props.children.toString();
      expect(textContent).toMatch(/^[a-zA-Z0-9.\s]*$/);
    });
  });

  describe("ExtraView", () => {
    test("Render correctly", () => {
      const { getByTestId } = signupScreenRender;
      const extraViewElement = getByTestId("extra-view");
      expect(extraViewElement).toBeTruthy();
    });
  });

  describe("ExtraText", () => {
    test("Render correctly", () => {
      const { getByTestId } = signupScreenRender;
      const extraTextElement = getByTestId("extra-text");
      expect(extraTextElement).toBeTruthy();
    });

    test("Render the correct text", () => {
      const { getByTestId } = signupScreenRender;
      const extraTextElement = getByTestId("extra-text");
      const textContent = extraTextElement.props.children;
      expect(textContent).toBe("Already have an account?");
    });
  });

  describe("TextLink", () => {
    test("Render correctly", () => {
      const { getByTestId } = signupScreenRender;
      const textLinkElement = getByTestId("text-link");
      expect(typeof textLinkElement.props).toBe("object");
    });

    test("Render the correct text", () => {
      const { getByTestId } = signupScreenRender;
      const textLinkContent = getByTestId("text-link-content");
      const LinkContent = textLinkContent.props.children;
      expect(LinkContent).toBe("Login");
    });
  });
});
