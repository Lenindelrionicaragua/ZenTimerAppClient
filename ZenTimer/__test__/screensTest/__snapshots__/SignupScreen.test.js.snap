// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupScreen Renders correctly the SignupScreen Component 1`] = `
<View
  style={
    {
      "backgroundColor": "#f9f7f3",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingBottom": 25,
      "paddingLeft": 25,
      "paddingRight": 25,
      "paddingTop": 64,
    }
  }
  testID="styled-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      }
    }
    testID="inner-container"
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 30,
          "fontWeight": "bold",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
      testID="page-title"
    >
      ZenTimer
    </Text>
    <Text
      style={
        {
          "color": "#1a1204",
          "fontSize": 10,
          "fontWeight": "bold",
          "letterSpacing": 1,
          "marginBottom": 20,
        }
      }
      testID="sub-title"
    >
      Account Signup
    </Text>
    <View
      style={
        {
          "width": "90%",
        }
      }
    >
      <View>
        <View
          style={
            {
              "left": 15,
              "position": "absolute",
              "top": 38,
              "zIndex": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                {
                  "fontFamily": "octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#1a1204",
              "fontSize": 13,
              "textAlign": "left",
            }
          }
        >
          Full Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          onblur={[Function]}
          placeholder="Zen User"
          placeholderTextColor="#f9f7f3"
          style={
            {
              "backgroundColor": "#ffc500",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "color": "#1a1204",
              "fontSize": 16,
              "height": 60,
              "marginBottom": 10,
              "marginVertical": 3,
              "paddingBottom": 15,
              "paddingLeft": 55,
              "paddingRight": 55,
              "paddingTop": 15,
            }
          }
          testID="full-name"
          value=""
        />
      </View>
      <View>
        <View
          style={
            {
              "left": 15,
              "position": "absolute",
              "top": 38,
              "zIndex": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                {
                  "fontFamily": "octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#1a1204",
              "fontSize": 13,
              "textAlign": "left",
            }
          }
        >
          Email Address
        </Text>
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          onblur={[Function]}
          placeholder="serenity@gmail.com"
          placeholderTextColor="#f9f7f3"
          style={
            {
              "backgroundColor": "#ffc500",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "color": "#1a1204",
              "fontSize": 16,
              "height": 60,
              "marginBottom": 10,
              "marginVertical": 3,
              "paddingBottom": 15,
              "paddingLeft": 55,
              "paddingRight": 55,
              "paddingTop": 15,
            }
          }
          testID="email-input"
          value=""
        />
      </View>
      <View>
        <View
          style={
            {
              "left": 15,
              "position": "absolute",
              "top": 38,
              "zIndex": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                {
                  "fontFamily": "octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#1a1204",
              "fontSize": 13,
              "textAlign": "left",
            }
          }
        >
          Date of Birth
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <TextInput
            editable={false}
            onChangeText={[Function]}
            onblur={[Function]}
            placeholder="YYYY - MM - DD"
            placeholderTextColor="#f9f7f3"
            style={
              {
                "backgroundColor": "#ffc500",
                "borderBottomLeftRadius": 5,
                "borderBottomRightRadius": 5,
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "color": "#1a1204",
                "fontSize": 16,
                "height": 60,
                "marginBottom": 10,
                "marginVertical": 3,
                "paddingBottom": 15,
                "paddingLeft": 55,
                "paddingRight": 55,
                "paddingTop": 15,
              }
            }
            testID="date-of-birth"
            value=""
          />
        </View>
      </View>
      <View>
        <View
          style={
            {
              "left": 15,
              "position": "absolute",
              "top": 38,
              "zIndex": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                {
                  "fontFamily": "octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#1a1204",
              "fontSize": 13,
              "textAlign": "left",
            }
          }
        >
          Password
        </Text>
        <TextInput
          onChangeText={[Function]}
          onblur={[Function]}
          placeholder="* * * * * * *"
          placeholderTextColor="#f9f7f3"
          secureTextEntry={true}
          style={
            {
              "backgroundColor": "#ffc500",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "color": "#1a1204",
              "fontSize": 16,
              "height": 60,
              "marginBottom": 10,
              "marginVertical": 3,
              "paddingBottom": 15,
              "paddingLeft": 55,
              "paddingRight": 55,
              "paddingTop": 15,
            }
          }
          testID="password-input"
          value=""
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "position": "absolute",
              "right": 15,
              "top": 38,
              "zIndex": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#f9f7f3",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View>
        <View
          style={
            {
              "left": 15,
              "position": "absolute",
              "top": 38,
              "zIndex": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                {
                  "fontFamily": "octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#1a1204",
              "fontSize": 13,
              "textAlign": "left",
            }
          }
        >
          Confirm Password
        </Text>
        <TextInput
          onChangeText={[Function]}
          onblur={[Function]}
          placeholder="* * * * * * *"
          placeholderTextColor="#f9f7f3"
          secureTextEntry={true}
          style={
            {
              "backgroundColor": "#ffc500",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "color": "#1a1204",
              "fontSize": 16,
              "height": 60,
              "marginBottom": 10,
              "marginVertical": 3,
              "paddingBottom": 15,
              "paddingLeft": 55,
              "paddingRight": 55,
              "paddingTop": 15,
            }
          }
          testID="confirm-password-input"
          value=""
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "position": "absolute",
              "right": 15,
              "top": 38,
              "zIndex": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#f9f7f3",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "fontSize": 13,
            "textAlign": "center",
          }
        }
        testID="msg-box"
      >
        ...
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#000000",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "height": 60,
            "justifyContent": "center",
            "marginVertical": 5,
            "opacity": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          }
        }
        testID="login-styled-button"
      >
        <Text
          style={
            {
              "color": "#ffe1a8",
              "fontSize": 16,
            }
          }
          testID="login-button-text"
        >
          Login
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#ffc500",
            "height": 1,
            "marginVertical": 10,
            "width": "100%",
          }
        }
        testID="line"
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        testID="extra-view"
      >
        <Text
          style={
            {
              "alignContent": "center",
              "color": "#1a1204",
              "fontSize": 15,
              "justifyContent": "center",
            }
          }
          testID="extra-text"
        >
          Already have an account?
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="text-link"
        >
          <Text
            style={
              {
                "color": "#ffc554",
                "fontSize": 15,
              }
            }
            testID="text-link-content"
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
